uniffi_bindgen/src/bindings/kotlin/templates/ObjectTemplate.kt

```diff
@@ -57,19 +57,43 @@ class {{ type_name }}(
     {%- if meth.is_async() %}
     @Suppress("ASSIGNED_BUT_NEVER_ACCESSED_VARIABLE")
     override suspend fun {{ meth.name()|fn_name }}({%- call kt::arg_list_decl(meth) -%}){% match meth.return_type() %}{% when Some with (return_type) %} : {{ return_type|type_name }}{% when None %}{%- endmatch %} {
-        // Create a new `CoroutineScope` for this operation, suspend the coroutine, and call the
-        // scaffolding function, passing it one of the callback handlers from `AsyncTypes.kt`.
-        //
-        // Make sure to retain a reference to the callback handler to ensure that it's not GCed before
-        // it's invoked
-        var callbackHolder: {{ func.result_type().borrow()|future_callback_handler }}? = null
+        {#
+        Create a new `CoroutineScope` for this operation, suspend the coroutine, and call the
+        scaffolding function, passing it one of the callback handlers from `AsyncTypes.kt`.
+
+        Make sure to retain a reference to the callback handler to ensure that it's not GCed before
+        it's invoked
+        #}
+        var callbackHolder: {{ meth.result_type().borrow()|future_callback_handler }}? = null
+        var rustFuturePtr: Pointer? = null
+        val completionHandlerLock = Mutex(locked = true)
+
         return coroutineScope {
             val scope = this
-            return@coroutineScope suspendCoroutine { continuation ->
+            val completionHandler: CompletionHandler = { _ ->
+                runBlocking {
+                    completionHandlerLock.withLock {
+                        if (rustFuturePtr != null) {
```

Comment:
```kotlin
rustFuturePtr?.let { ptr -> ... } 
```
Is more idiomatic and you avoid having to force unwrap inside the if.

Reply:
Solved, thanks.